Artroom1/ArtworksDatabase.swift

```diff
@@ -13,33 +13,57 @@ import UIKit
 
 public class ArtworksDatabase {
     static let shared = ArtworksDatabase()
-    var arrayOfArtworks: [Artwork]
-    
+    private(set) var arrayOfArtworks: [Artwork] // Because I only want to mutate it here but read it from the outside
+    private(set) var dictOfImages: [String: UIImage]
+
     private init() {
         arrayOfArtworks = []
+        dictOfImages = [:]
     }
-    
-    func load(completionHandler: @escaping () -> Void ) {
-        loadJson { (artworks, error) in
+
+    func load(completionHandler: @escaping () -> Void) {
+        loadJson { artworks, _ in
             if let artworks = artworks {
                 self.arrayOfArtworks = artworks
+                // In Progress: The following is supposed to create a dictionary of images, to get them ready for use later instead of making new network requests.
+                artworks.forEach({ artwork in
```

Comment:
Imagine one of the images is ~150 kb. Now the users are on a slow 3G connection and have to download 100 images before they can even start using the app. I think that is not a nice user experience :) Also the probability of an user seeing all images is quite low. I'd keep the code as before, so only downloading when the image is actually needed.